### ISSUE TITLE
[Question] Pretraining warmup steps / learning rate

### ISSUE THREAD

**Issue by kolakows** [google-research/pegasus#190]
Hi,

In the paper you were using adafactor with square root learning rate decay for pretraining. How many warmup steps did you have? And also lr = 0.1 seems rather high, did you encounter model divergence?

**Comment by JingqingZ** [comment]
Hi,

Warmup steps: 10k

https://github.com/google-research/pegasus/blob/13fcf2b1191e0df950436c82c33d672c1447f5ff/pegasus/params/estimator_utils.py#L112

The dropout rate is 0.1 but not lr.

**Comment by kolakows** [comment]
I took the lr from a table in appendix C, in the learning rate column for pegasus_base pretraining. I guess a typo.

Thanks for the code reference, I see that init_lr and lr are set to 0.01. 

There is also a warmup steps parameter in the code, but from what I can see, the adafactor optimizer doesn't take it as input. Internally when not supplied with lr, adafactor will use the formula from below, which uses warmup = 10k, but in the code lr is passed to the init as 0.01.

https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/adafactor.py#L57

But then again, the formula is only used once in the initialization (code below).

https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/adafactor.py#L148

I'm not familiar with the internal workings of the adafactor or tf optimizers, but from what i can see, it looks like lr doesn't depend on step number (unless the optimizer was initialized multiple times), code below

https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/adafactor.py#L221

Sorry for the long comment, and thanks for the clarification. I'll try the recommended 0.01 with warmup 10k, also 0.001 constant seems to be working fine. (btw. I'm using Huggingface for the training, so the learning rates might and are probably working differently)

**Comment by kolakows** [comment]
On second thought I guess the update depending on step number is done somewhere with a scheduler and I just couldn't find it.
